src/fast/u192.rs

```diff
@@ -0,0 +1,57 @@
+use std::mem::transmute;
+use std::ops::{Add, Mul};
+
+// This was started by the output of the uint crate,
+// then heavily reduced to only the parts we need
+// (which was a significant optimization, especially
+// in the mul which carried out the full multiplication
+// and discarded the top-half)
+//
+/// Little-endian large integer type
+#[derive(Copy, Clone, PartialEq, Eq, Hash, Debug)]
+pub struct U192(pub [u64; 3]);
+
+impl Mul for U192 {
+    type Output = Self;
+    fn mul(self, other: Self) -> Self {
+        // The generated implementation of this method was 360 lines long!
+        let me = &self.0;
+        let you = &other.0;
+
+        let mult = |m: usize, y: usize| unsafe {
+            transmute::<_, (u64, u64)>(me[m] as u128 * you[y] as u128)
```

Comment:
Doesn't that `transmute` assume something about the layout of a `u128`? The meaning of the first and second part of the result tuple will be different on little and big endian machines

Reply:
Yes, I got lazy here. The implementation is supposed to produce the same result across architectures but does not. I think I'll just ban big-endian machines by panicking somewhere for now since I cannot actually test with one and there is very little chance of an Indexer using that for graph-node.